src/main/java/io/strimzi/kafka/metrics/PrometheusMetricsReporterConfig.java

```diff
@@ -82,4 +134,62 @@ public synchronized Optional<HTTPServer> startHttpServer() {
         }
     }
 
+    static class ListenerValidator implements ConfigDef.Validator {
+
+        @Override
+         public void ensureValid(String name, Object value) {
+            Matcher matcher = Listener.PATTERN.matcher(String.valueOf(value));
+            if (!matcher.matches()) {
+                throw new ConfigException(name, value, "Listener must be of format http://[host]:[port]");
+            }
+        }
+    }
+
+    static class Listener {
+
+        private static final Pattern PATTERN = Pattern.compile("http://\\[?([0-9a-zA-Z\\-%._:]*)]?:([0-9]+)");
+
+        final String host;
+        final int port;
+
+        Listener(String host, int port) {
```

Comment:
I wonder if you actually need the Listener class at all.  You could just have parseListener return a InetSocketAddress....

Reply:
We considered that option, and I think it makes sense if we switch to use an `InetSocketAddress`. We decided to keep a container class as I expect we'll add https in the near future.

Reply:
We should probably have had these discussions in public so you (and others) could see our thought process.

Reply:
Does it make sense to keep the `Listener` class in with the prospect of adding https?

Reply:
@mimaison @k-wall I've added the InetSocketAddress but kept the Listener class. Everything looks to be working as expected though I am open to correction.